src/FakeItEasy/Configuration/RuleBuilder.cs

```diff
@@ -141,11 +142,21 @@ public UnorderedCallAssertion MustHaveHappened(Repeated repeatConstraint)
         {
             Guard.AgainstNull(repeatConstraint, nameof(repeatConstraint));
 
-            var asserter = this.asserterFactory.Invoke(this.Calls);
+            return this.MustHaveHappened(repeatConstraint.ToCallCountConstraint());
+        }
 
-            asserter.AssertWasCalled(this.Matcher.Matches, this.RuleBeingBuilt.WriteDescriptionOfValidCall, repeatConstraint);
+        public UnorderedCallAssertion MustHaveHappened(int numberOfTimes, Times timesOption)
+        {
+            Guard.AgainstNull(timesOption, nameof(timesOption));
 
-            return new UnorderedCallAssertion(this.manager, this.Matcher, this.RuleBeingBuilt.WriteDescriptionOfValidCall, repeatConstraint);
+            return this.MustHaveHappened(timesOption.ToCallCountConstraint(numberOfTimes));
+        }
+
+        public UnorderedCallAssertion MustHaveHappenedANumberOfTimesMatching(Expression<Func<int, bool>> predicate)
+        {
+            Guard.AgainstNull(predicate, nameof(predicate));
+
+            return this.MustHaveHappened(new CallCountConstraint(predicate.Compile(), $"the number of times specified by the predicate '{predicate}'"));

```

Comment:
Maybe _"a number of times matching the predicate"_ instead of _"the number of times specified by the predicate"_? It would be more accurate, and would fit the API language better.

Reply:
Oh, fantastic observation, @thomaslevesque. Tell you what. I'll bundle [a slightly modified version of] it in with another commit that I was planning on sending anyhow. I look forward to your reaction.

Reply:
Very nice! 👍